nixos/modules/services/misc/autorandr.nix

```diff
@@ -22,22 +22,16 @@ in {
 
     environment.systemPackages = [ pkgs.autorandr ];
 
-    # systemd.unitPackages = [ pkgs.autorandr ];
+    systemd.packages = [ pkgs.autorandr ];
+
     systemd.services.autorandr = {
-      unitConfig = {
-        Description = "autorandr execution hook";
-        After = [ "sleep.target" ];
-        StartLimitInterval = "5";
-        StartLimitBurst = "1";
-      };
-      serviceConfig = {
-        ExecStart = "${pkgs.autorandr}/bin/autorandr --batch --change --default default";
-        Type = "oneshot";
-        RemainAfterExit = false;
-      };
       wantedBy = [ "sleep.target" ];
     };
 
+    services.udev.extraRules = ''
+      ACTION=="change", SUBSYSTEM=="drm", RUN+="${pkgs.bash}/bin/sh -c '${pkgs.systemd}/bin/systemctl is-active -q graphical.target && ${pkgs.systemd}/bin/systemctl start autorandr.service'"
```

Comment:
Then I would rename the upstream unit to `autorandr@.service` and use `BindsTo` of systemd
to make the service restart, whenever the device appears/disappears:

```
ACTION=="change", SUBSYSTEM=="drm", TAG+="systemd", ENV{SYSTEMD_WANTS}="autorandr@%E{DEVNAME}.service"
```

If this does also not work. I would at least remove the raise condition (systemctl is-active check) and make systemctl non-blocking:

 `${pkgs.systemd}/bin/systemctl start --no-block autorandr.service`